src/plugins/open/open.c

```diff
@@ -353,71 +368,80 @@ ipmi_openipmi_send_cmd(struct ipmi_intf * intf, struct ipmi_rq * req)
 	recv.msg.data = rsp.data;
 	recv.msg.data_len = sizeof(rsp.data);
 
 	/* get data */
 	if (ioctl(intf->fd, IPMICTL_RECEIVE_MSG_TRUNC, &recv) < 0) {
-	   lperror(LOG_ERR, "Error receiving message");
-	   if (errno != EMSGSIZE) {
-	      if (data) {
-					free(data);
-					data = NULL;
-				}
-	      return NULL;
-	   }
+		lperror(LOG_ERR, "Error receiving message");
+		if (errno != EMSGSIZE) {
+			if (data) {
+				free(data);
+				data = NULL;
+			}
+			return NULL;
+		}
 	}
 
 	if (verbose > 4) {
-	   fprintf(stderr, "Got message:");
-	   fprintf(stderr, "  type      = %d\n", recv.recv_type);
-	   fprintf(stderr, "  channel   = 0x%x\n", addr.channel);
-	   fprintf(stderr, "  msgid     = %ld\n", recv.msgid);
-	   fprintf(stderr, "  netfn     = 0x%x\n", recv.msg.netfn);
-	   fprintf(stderr, "  cmd       = 0x%x\n", recv.msg.cmd);
-	   if (recv.msg.data && recv.msg.data_len) {
-	      fprintf(stderr, "  data_len  = %d\n", recv.msg.data_len);
-	      fprintf(stderr, "  data      = %s\n",
-		      buf2str(recv.msg.data, recv.msg.data_len));
-	   }
+		fprintf(stderr, "Got message:");
+		fprintf(stderr, "  type      = %d\n", recv.recv_type);
+		fprintf(stderr, "  channel   = 0x%x\n", addr.channel);
+		fprintf(stderr, "  msgid     = %ld\n", recv.msgid);
+		fprintf(stderr, "  netfn     = 0x%x\n", recv.msg.netfn);
+		fprintf(stderr, "  cmd       = 0x%x\n", recv.msg.cmd);
+		if (recv.msg.data && recv.msg.data_len) {
+			fprintf(stderr, "  data_len  = %d\n",
+				recv.msg.data_len);
+			fprintf(stderr, "  data      = %s\n",
+				buf2str(recv.msg.data, recv.msg.data_len));
+		}
 	}
 
-	if(intf->transit_addr != 0 && intf->transit_addr != intf->my_addr) {
-	   /* ipmb_addr.transit_slave_addr = intf->transit_addr; */
-	   lprintf(LOG_DEBUG, "Decapsulating data received from transit "
-		   "IPMB target @ 0x%x", intf->transit_addr);
-
-	   /* comp code */
-	   /* Check data */
-
-	   if( recv.msg.data[0] == 0 ) {
-	      recv.msg.netfn = recv.msg.data[2] >> 2;
-	      recv.msg.cmd   = recv.msg.data[6];
-
-	      recv.msg.data = memmove(recv.msg.data ,recv.msg.data+7 , recv.msg.data_len - 7);
-	      recv.msg.data_len -=8;
-       
-	      if (verbose > 4) {   
-		 fprintf(stderr, "Decapsulated  message:\n");
-		 fprintf(stderr, "  netfn     = 0x%x\n",   recv.msg.netfn );
-		 fprintf(stderr, "  cmd       = 0x%x\n",  recv.msg.cmd);
-		 if (recv.msg.data && recv.msg.data_len) {
-		    fprintf(stderr, "  data_len  = %d\n",  recv.msg.data_len);
-		    fprintf(stderr, "  data      = %s\n",
-			    buf2str(recv.msg.data,recv.msg.data_len));
-		 }
-	      }
-	   }
+	if (intf->transit_addr != 0 && intf->transit_addr != intf->my_addr) {
+		/* ipmb_addr.transit_slave_addr = intf->transit_addr; */
+		lprintf(LOG_DEBUG,
+			"Decapsulating data received from transit "
+			"IPMB target @ 0x%x",
+			intf->transit_addr);
+
+		/* comp code */
+		/* Check data */
+
+		if (recv.msg.data[0] == 0) {
+			recv.msg.netfn = recv.msg.data[2] >> 2;
+			recv.msg.cmd = recv.msg.data[6];
```

Comment:
I agree, can it be pushed onto an issue or another PR, I see IPMI_OK could be used instead of 0, since that's what they're really checking.  And then a few macros to get the command out, etc.